config_utilities/src/asl_formatter.cpp

```diff
@@ -33,11 +33,24 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  * -------------------------------------------------------------------------- */
 
+#include "config_utilities/factory.h"
 #include "config_utilities/formatting/asl.h"
 #include "config_utilities/internal/string_utils.h"
 #include "config_utilities/settings.h"
 
 namespace config::internal {
+namespace {
+
+// Factory registration to allow setting of formatters via Settings::setFormatter().
+static const auto registration = Registration<Formatter, AslFormatter>("asl");
+
+}  // namespace
+
+AslFormatter::AslFormatter() = default;
```

Comment:
Is there a reason for defaulting constructors in the src file? I'm more used to seeing that in the header (and vice versa, if I see a constructor declaration I assume it's non-default)